ui/src/app/views/concept-table/component.tsx

```diff
@@ -135,10 +143,35 @@ export class ConceptTable extends React.Component<{concepts: Concept[];
 
   rowExpansionTemplate(data) {
     return (<SynonymsObject>
-      {fp.uniq(data.conceptSynonyms).join(', ')}
+      {this.highlightWithSearchTerm(fp.uniq(data.conceptSynonyms).join(', '))}
     </SynonymsObject>);
   }
 
+  highlightWithSearchTerm(stringToHighlight: string) {
+    const {searchTerm} = this.props;
+    if (searchTerm) {
+      const words: string[] = [];
+      let searchWords = searchTerm.split(new RegExp(',| '));
+      searchWords = searchWords.filter(w => w.length > 0 );
+      searchWords = searchWords.map(word => word.replace(/[&!^\/\\#,+()$~%.'":*?<>{}]/g, ''));
```

Comment:
Can we write this as searchTerm.split(new RegExp(',| '))
        .filter(w => w.length > 0 )
        .map(word => word.replace(/[&!^\/\\#,+()$~%.'":*?<>{}]/g, ''))?